module-apps/application-messages/windows/SMSTemplatesWindow.cpp

```diff
@@ -116,8 +116,21 @@ namespace gui
         }
 
         if (auto switchData = dynamic_cast<SMSSendTemplateRequest *>(data); switchData != nullptr) {
+            ignoreWindowsOfThisAppOnSwitchBack = data->ignoreCurrentWindowOnStack;
+            appNameToSwitchBack                = switchData->getNameOfSenderApp();
             smsSendTemplateRequestHandler(switchData);
         }
     }
 
+    bool SMSTemplatesWindow::onInput(const InputEvent &inputEvent)
+    {
+        if (inputEvent.isShortRelease(KeyCode::KEY_RF) && ignoreWindowsOfThisAppOnSwitchBack &&
```

Comment:
You could negate this conditions and return early - then, the code would gain some readability due to less brackets & indents.